packages/common/src/logging/config.ts

```diff
@@ -0,0 +1,358 @@
+  INFO = 2,
+  /** Debug messages and all above */
+  DEBUG = 3,
+  /** All messages including verbose details */
+  VERBOSE = 4,
+}
+
+/**
+ * Configuration for redacting sensitive data in log output.
+ */
+export type RedactConfig = {
+  /** Predicate function to test if a key should be redacted. Returns true if the key should be redacted. */
+  keys?: (key: string) => boolean;
+  /** Function to replace redacted values. Defaults to returning "[REDACTED]". */
+  replacer?: (value: unknown, key?: string) => unknown;
+};
+
+/**
+ * Configuration for the diagnostics logging system.
+ * All properties are optional and will use sensible defaults if not provided.
+ */
+export type DiagnosticsConfig = {
+  /** Minimum log level to emit. Defaults to INFO in development, ERROR in production. */
+  level?: LogLevel;
+  /** Configuration for redacting sensitive data in logs. */
+  redact?: RedactConfig;
+  /** Array of sink functions to receive log entries. Defaults to console sink. */
+  sinks?: Array<(entry: LogEntry) => void>;
+  /** Disable telemetry headers in requests. Defaults to false. */
+  disableTelemetryHeaders?: boolean;
+  /** Array of exact namespace strings to allow. If undefined or empty, all namespaces are enabled. Example: ["aa-infra", "wallet-apis"] */
+  enabledNamespaces?: string[];
+};
+
+/**
+ * Base context attached to all log entries from a logger instance.
+ */
+export type LoggerContextBase = {
+  /** Package name (e.g., "@alchemy/aa-infra") */
+  package: string;
+  /** Package version (e.g., "1.0.0") */
+  version: string;
+};
+
+/**
+ * A single log entry emitted by the diagnostics logger.
+ */
+export type LogEntry = {
+  /** Timestamp in milliseconds since epoch (Date.now()) */
+  ts: number;
+  /** Log level of this entry */
+  level: LogLevel;
+  /** Optional namespace for filtering/grouping (e.g., "aa-infra", "wallet-apis") */
+  namespace: string | undefined;
+  /** The log message */
+  message: string;
+  /** Optional structured data attached to this log entry */
+  data?: Record<string, unknown>;
+  /** Package context (name and version) */
+  context: LoggerContextBase;
+};
+
+let globalConfig: Required<DiagnosticsConfig> | undefined;
+
+function defaultConfig(): Required<DiagnosticsConfig> {
+  const env = (typeof process !== "undefined" && process.env) || {};
+
+  // Default level: INFO in dev, ERROR in prod
+  const isDev = env.NODE_ENV === "development";
+  const defaultLevel = isDev ? LogLevel.INFO : LogLevel.ERROR;
+
+  let level = defaultLevel;
+  const envLevel = env.ALCHEMY_LOG_LEVEL;
+  if (envLevel) {
+    const map: Record<string, LogLevel> = {
+      error: LogLevel.ERROR,
+      warn: LogLevel.WARN,
+      info: LogLevel.INFO,
+      debug: LogLevel.DEBUG,
+      verbose: LogLevel.VERBOSE,
+    };
+    const normalized = envLevel.toLowerCase();
+    if (normalized in map) {
+      level = map[normalized];
+    } else {
+      const asNum = Number(envLevel);
```

Comment:
do we really need to support both numbers and strings for the env level? imo we can simplify this and just use the strings.

if we do that & stop using the `enum`, we can simply do this here:

```ts
LogLevel[envLevel.toUpperCase() as keyof typeof LogLevel];
```